### PR TITLE
Feature/add error history module

### PR DESCRIPTION
**Pull request by andistorm**
requires:
* https://github.com/EVerest/everest-utils/pull/79
* https://github.com/EVerest/everest-core/pull/446
* https://github.com/EVerest/everest-framework/pull/133

does not require any longer:
* https://github.com/EVerest/everest-utils/pull/110
* https://github.com/EVerest/everest-ci/pull/10

### CODE REVIEW

**Comment by andistorm** [comment]
Check required ec-cli before merging

**Comment by andistorm** [comment]
Do not merge before setting correct dependencies in dependencies.yaml

**Comment by corneliusclaussen** [comment]
Converting to draft as it is waiting for other PRs and cannot be merged now. Also needs cleanup of history after dependent PRs are merged.

File: dependencies.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 everest-framework:
   git: https://github.com/EVerest/everest-framework.git
-  git_tag: v0.9.0
+  git_tag: refactor/error-handling
```
Comment (andistorm) [inline dependencies.yaml:4]:
```suggestion
  git_tag: v0.9.1
```

File: dependencies.yaml [hunk 2]
```diff
@@ -66,4 +66,11 @@ everest-utils:
 # setting it here can be misleading since it does not affect the version being used
 libevse-security:
   git: https://github.com/EVerest/libevse-security.git
-  git_tag: 1604c8b
+  git_tag: v0.4.1
```
Comment (andistorm) [inline dependencies.yaml:69]:
```suggestion
  git_tag: 1604c8b
```

File: dependencies.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ---
 everest-framework:
   git: https://github.com/EVerest/everest-framework.git
-  git_tag: v0.9.0
+  git_tag: v0.9.1
```
Comment (andistorm) [inline dependencies.yaml:4]:
```suggestion
  git_tag: v0.10.0
```

**Review by hikinggrass — APPROVED** [review]

File: modules/ErrorHistory/tests/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,28 @@
+add_executable(module_error_history_tests)
+
```
Comment (andistorm) [inline modules/ErrorHistory/tests/CMakeLists.txt:2]:
```suggestion
set(TARGET_NAME ${PROJECT_NAME}_module_error_history_tests)
add_executable(${TARGET_NAME})

```

File: modules/ErrorHistory/tests/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+target_sources(module_error_history_tests
```
Comment (andistorm) [inline modules/ErrorHistory/tests/CMakeLists.txt:3]:
```suggestion
target_sources(${TARGET_NAME}
```

File: modules/ErrorHistory/tests/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        error_database_sqlite_tests.cpp
+        ../ErrorDatabaseSqlite.cpp
+        helpers.cpp
+)
+
+# target_include_directories(module_error_history_tests
+#     PRIVATE
+# )
+
+target_link_libraries(module_error_history_tests
```
Comment (andistorm) [inline modules/ErrorHistory/tests/CMakeLists.txt:14]:
```suggestion
target_link_libraries(${TARGET_NAME}
```

File: modules/ErrorHistory/tests/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+        everest::framework
+        everest::log
+        SQLiteCpp
+        SQLite::SQLite3
+        Catch2::Catch2WithMain
+)
+if(NOT DISABLE_EDM)
+    list(APPEND CMAKE_MODULE_PATH ${CPM_PACKAGE_catch2_SOURCE_DIR}/extras)
+    include(Catch)
+    catch_discover_tests(module_error_history_tests)
```
Comment (andistorm) [inline modules/ErrorHistory/tests/CMakeLists.txt:25]:
```suggestion
    catch_discover_tests(${TARGET_NAME})
```

File: modules/ErrorHistory/CMakeLists.txt [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+)
+# ev@bcc62523-e22b-41d7-ba2f-825b493a3c97:v1
+
+target_sources(${MODULE_NAME}
+    PRIVATE
+        "error_history/error_historyImpl.cpp"
+)
+
+# ev@c55432ab-152c-45a9-9d2e-7281d50c69c3:v1
+if(BUILD_TESTING)
```
Comment (andistorm) [inline modules/ErrorHistory/CMakeLists.txt:36]:
```suggestion
if(EVEREST_CORE_BUILD_TESTING)
```

File: modules/ErrorHistory/tests/CMakeLists.txt [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+        SQLite::SQLite3
+        Catch2::Catch2WithMain
+)
+if(NOT DISABLE_EDM)
+    list(APPEND CMAKE_MODULE_PATH ${CPM_PACKAGE_catch2_SOURCE_DIR}/extras)
+    include(Catch)
+    catch_discover_tests(${TARGET_NAME})
+endif()
+
+add_test(module_error_history_tests module_error_history_tests)
```
Comment (andistorm) [inline modules/ErrorHistory/tests/CMakeLists.txt:29]:
```suggestion
add_test(${TARGET_NAME} ${TARGET_NAME})
```

**Pull request merged**